### PR TITLE
removed reflection and use new public API from metadata reference

### PR DESCRIPTION
**Pull request by heejaechang**
I used reflection to get byte array from metadata reference before. now these are exposed as public api from metadata. so removed reflection and using public api directly.

### CODE REVIEW

**Comment by heejaechang** [comment]
@tmat can you take a look? @dotnet/roslyn-ide @dotnet/roslyn-analysis

**Comment by heejaechang** [comment]
retest this please

test seems stuck

**Comment by heejaechang** [comment]
ping? @tmat it is very simple PR.

File: src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs [hunk 1]
```diff
@@ -215,16 +215,10 @@ private void WriteMvidTo(ModuleMetadata metadata, ObjectWriter writer, Cancellat
...
-            // TODO: what I should do with documentation provider? it is not exposed outside
-        }
+            WriteTo(reference.GetMetadata(), writer, cancellationToken);
 
-        private void WritePortableExecutableReferenceMetadataTo(
-            PortableExecutableReference reference, ObjectWriter writer, CancellationToken cancellationToken)
-        {
-            var metadata = GetMetadata(reference);
-            WriteTo(metadata, writer, cancellationToken);
+            // TODO: what I should do with documentation provider? it is not exposed outside
```
Comment (tmat) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: Should we have an issue for this TODO?

Comment (heejaechang) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: I already have one. the comment is not new.

Comment (tmat) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: Perhaps add a link to the issue?

Comment (heejaechang) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: I have issue with bunch of TODOs in them. I am actively going them one by one. so I don't think for now I need link into the code yet.

things I decide not to fix in short term I will add those link.

File: src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs [hunk 2]
```diff
@@ -456,37 +450,20 @@ private void WriteTo(ModuleMetadata metadata, ObjectWriter writer, CancellationT
...
             cancellationToken.ThrowIfCancellationRequested();
 
-            var blockFieldInfo = reader.GetType().GetTypeInfo().GetDeclaredField("Block");
-            var block = blockFieldInfo.GetValue(reader);
+            var length = reader.MetadataLength;
 
-            // TODO: once things become public API, change it to copy stream over byte* and length from metadata reader
-            var toArrayFieldInfo = block.GetType().GetTypeInfo().GetDeclaredMethod("ToArray");
-            var array = (byte[])toArrayFieldInfo.Invoke(block, null);
+            // TODO: any way to avoid allocating byte array here?
```
Comment (tmat) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: Issue?

**Comment by tmat** [comment]
:+1:

File: src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs (hunk 2 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: will do. for now, added to this. https://github.com/dotnet/roslyn/issues/12770

**Pull request merged**